app/components/retriable_fast_image/index.tsx

```diff
@@ -2,40 +2,86 @@
 // See LICENSE.txt for license information.
 
 import React, {PureComponent} from 'react';
+import CompassIcon from '@components/compass_icon';
 import FastImage, {FastImageProps} from 'react-native-fast-image';
+import validator from 'validator';
+import {StyleSheet, View} from 'react-native';
 
 export const FAST_IMAGE_MAX_RETRIES = 3;
 
-type RetriableFastImageProps = FastImageProps & {
-    id: string
+interface RetriableFastImageState {
+    retries: number,
 }
 
-type RetriableFastImageState = {
-    retry: number
+interface RetriableFastImageProps extends FastImageProps {
+    id: string
+    retry?: boolean
+    renderErrorImage?: boolean
+    errorImageSize?: number
 }
 
 export default class RetriableFastImage extends PureComponent<RetriableFastImageProps, RetriableFastImageState> {
     state = {
-        retry: 0,
+        retries: 0,
     }
 
+    config: validator.IsURLOptions = {
+        protocols: ['file', 'http', 'https', 'content', 'data'],
+        require_host: false,
+        require_tld: false,
+        require_valid_protocol: true,
+        require_protocol: true,
+        allow_underscores: true,
+        allow_trailing_dot: true,
+    };
+
+    // Checks if given URI is valid
+    isValidUri = this.props.source && typeof this.props.source != 'number' && this.props.source.uri ? validator.isURL(this.props.source.uri, this.config) || validator.isDataURI(this.props.source.uri) : false;
```

Comment:
I'm not seeing anything triggering the `onError` prop if `isValidUrl` equals to false as this won't be rendering the FastImage component.

Reply:
Yup I explain a bit [here](https://github.com/mattermost/mattermost-mobile/pull/5345/files#r643557941) - I made the assumption that `this.props.onError` was initially designed to be run inside the `<FastImage onError={...}` so the methods being passed in might be specific to that; hence keeping `isValidUri` separate and only using that to display the broken image (if that prop is set).